xarray/core/groupby.py

```diff
@@ -370,8 +370,11 @@ def __init__(
                 group = group.dropna(group_dim)
 
             # look through group to find the unique values
+            sort = bins is None and (
+                not isinstance(safe_cast_to_index(group), pd.MultiIndex)
+            )
             unique_values, group_indices = unique_value_groups(
-                safe_cast_to_index(group), sort=(bins is None)
+                safe_cast_to_index(group), sort=sort
```

Comment:
I'm actually not sure why we generally sort; lmk if you happened to have found the reason when testing...

Reply:
So that was my first try, to simply remove the sorting, but then there were three test failures in `test_dataarray`, see e.g. [here](https://dev.azure.com/xarray/xarray/_build/results?buildId=2476&view=logs&j=ecaa9258-9d16-5560-78f9-8d2abe9f100f&t=df1f0ef6-5536-50dc-6554-aba88dd9b288&l=154).

They all involve grouping on 1D Indexes rather than MultiIndexes, which is what led me to the fix I settled on.  But that's as far as my understanding goes!